### DISCUSSION TITLE
How to invalidate every single query on Sign Out?

### DISCUSSION THREAD

**Discussion by apoorvmote** [TanStack/query#2274] [Q&A] [answered]
Most of my query keys are specific UUID that are used inside the query. 

On Sign Out page I don't have access to all the UUID that were used to make the query. 

There is one possibility that new signed in user will have different UUID so I don't need to worry about invalidating it. 

But I would like global invalidating queries without passing in specific query key.

**Comment by TkDodo** [comment, marked as answer]
You can do `queryClient.invalidateQueries()` without patenteres and it weil invalidate all of them. Also have a look at fuzzy query key matching: https://react-query.tanstack.com/guides/filters#query-filters

on logout, you might want removeQueries rather than invalidateQueries though

**Reply by apoorvmote** [reply]
Does the `queryClient.removeQueries()` work without passing in any parameters? This would be better for logout.

**Reply by TkDodo** [reply]
Yes, it does

**Reply by benadamstyles** [reply]
Is this the same as calling `queryClient.clear()`?

**Reply by TkDodo** [reply]
yes, pretty much. `clear` removes all queries from the queryCache, but also all mutations from the mutationCache.
